posts/2024-05-13-shinyMobile-2.0.0/index.qmd

```diff
@@ -0,0 +1,1868 @@
+              media = f7Icon("house"),
+              clearable = TRUE,
+              floating = TRUE
+            )
+          ),
+          f7TextArea(
+            inputId = "textarea",
+            label = "Text Area",
+            value = "Lorem ipsum dolor sit amet, consectetur
+              adipiscing elit, sed do eiusmod tempor incididunt ut
+              labore et dolore magna aliqua",
+            placeholder = "Your text here",
+            resize = TRUE,
+            style = list(
+              description = "A cool text input",
+              outline = TRUE,
+              media = f7Icon("house"),
+              clearable = TRUE,
+              floating = TRUE
+            )
+          ),
+          f7Password(
+            inputId = "password",
+            label = "Password:",
+            placeholder = "Your password here",
+            style = list(
+              description = "A cool text input",
+              outline = TRUE,
+              media = f7Icon("house"),
+              clearable = TRUE,
+              floating = TRUE
+            )
+          )
+        )
+      ),
+      verbatimTextOutput("form")
+    )
+  ),
+  server = function(input, output, session) {
+    output$form <- renderPrint(input$myform)
+
+    observeEvent(input$update, {
+      updateF7Form(
+        "myform",
+        data = list(
+          "text" = "New text",
+          "textarea" = "New text area",
+          "password" = "New password"
+        )
+      )
+    })
+  }
+)
+```
+
+## Breaking changes
+
+Some components have disappeared from Framework7 and we had to deprecate them as they no longer work. Other long time deprecated `{shinyMobile}` elements are also removed to cleanup the codebase. We invite you to review the [changelog](https://github.com/RinteRface/shinyMobile/blob/69da6ca46984bf6c73e2dc32ff8d9f415ec36a30/NEWS.md#breaking-changes) to see a list of all changes in this release.
+
+## Soft deprecation
+
+Some function parameters have changed and are now deprecated with `{lifecycle}`. You'll see a warning message if you use them and we invite you to accordingly update your code.
+
+# Conclusion
+
+Since the release of `{shiny}` in 2012, many packages substantially improve the layout and design such as `{bslib}`, `{bs4Dash}` or `{shiny.fluent}`. Yet not much was done for mobile development. `{shinyMobile}` tries to fill this gap by exposing people to the rich Framework7 mobile-first template. Coupled to progressive web app support (PWA), you can run Shiny apps on a mobile that look very close to native apps with a desktop icon, a launch screen and running fullscreen, that is without the web browser navigation bar. By including an experimental implementation of the multipage navigation as described earlier, we move one step closer to the native apps. Finally, owing to the progress on the [webR](https://docs.r-wasm.org/webr/latest/) side, it isn't impossible that one day, we might run a totally offline Shiny app on mobile.
```

Comment:
"Since the release of `{shiny}` in 2012, many packages **have been released** that substantially improve ~~the~~ **its** layout and design such as `{bslib}`, `{bs4Dash}` or `{shiny.fluent}`. Yet, not much was done for mobile development. `{shinyMobile}` tries to fill this gap by exposing people to the rich Framework7 mobile-first template. Coupled ~~to~~ **with** progressive web app support (PWA)..."

Reply:
Done. Thanks!